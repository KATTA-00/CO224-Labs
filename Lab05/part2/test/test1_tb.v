`include "test1.v"

module tb;

    reg [7:0] IN;
    reg [2:0]  INADDRESS, OUT1ADDRESS, OUT2ADDRESS;
    reg  WRITE, CLK, RESET;

    wire [7:0] OUT1, OUT2;

    reg_file file(IN,OUT1,OUT2,INADDRESS,OUT1ADDRESS,OUT2ADDRESS, WRITE, CLK, RESET);

    initial begin
        IN = 0; INADDRESS = 0; OUT1ADDRESS = 0; OUT2ADDRESS = 1; WRITE = 0; CLK = 0; RESET = 0;

        #3
        WRITE = 1;
        IN = 10;
        RESET = 0;

    end

    always
    #10 CLK = ~CLK;

    initial
    #50 $finish;

    initial begin
        $monitor($time, "%d | %d  %d | inA %d OutA %d %d | %d %d %d",IN,OUT1,OUT2,INADDRESS,OUT1ADDRESS,OUT2ADDRESS, WRITE, CLK, RESET);
    end




endmodule