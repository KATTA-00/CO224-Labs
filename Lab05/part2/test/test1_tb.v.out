#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "F:\iverilog\lib\ivl\system.vpi";
:vpi_module "F:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "F:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "F:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "F:\iverilog\lib\ivl\va_math.vpi";
S_00000172e2bce890 .scope module, "tb" "tb" 2 3;
 .timescale 0 0;
v00000172e2bb8420_0 .var "CLK", 0 0;
v00000172e2c23a60_0 .var "IN", 7 0;
v00000172e2c23ec0_0 .var "INADDRESS", 2 0;
v00000172e2c24320_0 .net "OUT1", 7 0, v00000172e2bb8920_0;  1 drivers
v00000172e2c248c0_0 .var "OUT1ADDRESS", 2 0;
v00000172e2c24820_0 .net "OUT2", 7 0, v00000172e2bb91e0_0;  1 drivers
v00000172e2c25220_0 .var "OUT2ADDRESS", 2 0;
v00000172e2c245a0_0 .var "RESET", 0 0;
v00000172e2c24640_0 .var "WRITE", 0 0;
S_00000172e2bcea20 .scope module, "file" "reg_file" 2 11, 3 1 0, S_00000172e2bce890;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "IN";
    .port_info 1 /OUTPUT 8 "OUT1";
    .port_info 2 /OUTPUT 8 "OUT2";
    .port_info 3 /INPUT 3 "INADDRESS";
    .port_info 4 /INPUT 3 "OUT1ADDRESS";
    .port_info 5 /INPUT 3 "OUT2ADDRESS";
    .port_info 6 /INPUT 1 "WRITE";
    .port_info 7 /INPUT 1 "CLK";
    .port_info 8 /INPUT 1 "RESET";
v00000172e2bb8d80_0 .net "CLK", 0 0, v00000172e2bb8420_0;  1 drivers
v00000172e2bb90a0_0 .net "IN", 7 0, v00000172e2c23a60_0;  1 drivers
v00000172e2bb8f60_0 .net "INADDRESS", 2 0, v00000172e2c23ec0_0;  1 drivers
v00000172e2bb8920_0 .var "OUT1", 7 0;
v00000172e2bb8560_0 .net "OUT1ADDRESS", 2 0, v00000172e2c248c0_0;  1 drivers
v00000172e2bb91e0_0 .var "OUT2", 7 0;
v00000172e2bb89c0_0 .net "OUT2ADDRESS", 2 0, v00000172e2c25220_0;  1 drivers
v00000172e2bb8ba0_0 .net "RESET", 0 0, v00000172e2c245a0_0;  1 drivers
v00000172e2bb8a60_0 .net "WRITE", 0 0, v00000172e2c24640_0;  1 drivers
v00000172e2bb8600_0 .var "register0", 7 0;
v00000172e2bb8c40_0 .var "register1", 7 0;
v00000172e2bb8ce0_0 .var "register2", 7 0;
v00000172e2bb8e20_0 .var "register3", 7 0;
v00000172e2bb9140_0 .var "register4", 7 0;
v00000172e2bb8ec0_0 .var "register5", 7 0;
v00000172e2bb8740_0 .var "register6", 7 0;
v00000172e2bb8380_0 .var "register7", 7 0;
E_00000172e2bba210 .event posedge, v00000172e2bb8d80_0;
E_00000172e2bba310/0 .event anyedge, v00000172e2bb8380_0, v00000172e2bb8740_0, v00000172e2bb8ec0_0, v00000172e2bb9140_0;
E_00000172e2bba310/1 .event anyedge, v00000172e2bb8e20_0, v00000172e2bb8ce0_0, v00000172e2bb8c40_0, v00000172e2bb8600_0;
E_00000172e2bba310/2 .event anyedge, v00000172e2bb89c0_0;
E_00000172e2bba310 .event/or E_00000172e2bba310/0, E_00000172e2bba310/1, E_00000172e2bba310/2;
E_00000172e2bb9690/0 .event anyedge, v00000172e2bb8380_0, v00000172e2bb8740_0, v00000172e2bb8ec0_0, v00000172e2bb9140_0;
E_00000172e2bb9690/1 .event anyedge, v00000172e2bb8e20_0, v00000172e2bb8ce0_0, v00000172e2bb8c40_0, v00000172e2bb8600_0;
E_00000172e2bb9690/2 .event anyedge, v00000172e2bb8560_0;
E_00000172e2bb9690 .event/or E_00000172e2bb9690/0, E_00000172e2bb9690/1, E_00000172e2bb9690/2;
    .scope S_00000172e2bcea20;
T_0 ;
    %wait E_00000172e2bb9690;
    %load/vec4 v00000172e2bb8560_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 3;
    %cmp/u;
    %jmp/1 T_0.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 3;
    %cmp/u;
    %jmp/1 T_0.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 3;
    %cmp/u;
    %jmp/1 T_0.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 3;
    %cmp/u;
    %jmp/1 T_0.3, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 3;
    %cmp/u;
    %jmp/1 T_0.4, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 3;
    %cmp/u;
    %jmp/1 T_0.5, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 3;
    %cmp/u;
    %jmp/1 T_0.6, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 3;
    %cmp/u;
    %jmp/1 T_0.7, 6;
    %delay 2, 0;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.0 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8600_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.1 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8c40_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.2 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8ce0_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.3 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8e20_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.4 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb9140_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.5 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8ec0_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.6 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8740_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.7 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8380_0;
    %store/vec4 v00000172e2bb8920_0, 0, 8;
    %jmp T_0.9;
T_0.9 ;
    %pop/vec4 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_00000172e2bcea20;
T_1 ;
    %wait E_00000172e2bba310;
    %load/vec4 v00000172e2bb89c0_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 3;
    %cmp/u;
    %jmp/1 T_1.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 3;
    %cmp/u;
    %jmp/1 T_1.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 3;
    %cmp/u;
    %jmp/1 T_1.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 3;
    %cmp/u;
    %jmp/1 T_1.3, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 3;
    %cmp/u;
    %jmp/1 T_1.4, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 3;
    %cmp/u;
    %jmp/1 T_1.5, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 3;
    %cmp/u;
    %jmp/1 T_1.6, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 3;
    %cmp/u;
    %jmp/1 T_1.7, 6;
    %delay 2, 0;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.0 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8600_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.1 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8c40_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.2 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8ce0_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.3 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8e20_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.4 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb9140_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.5 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8ec0_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.6 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8740_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.7 ;
    %delay 2, 0;
    %load/vec4 v00000172e2bb8380_0;
    %store/vec4 v00000172e2bb91e0_0, 0, 8;
    %jmp T_1.9;
T_1.9 ;
    %pop/vec4 1;
    %jmp T_1;
    .thread T_1, $push;
    .scope S_00000172e2bcea20;
T_2 ;
    %wait E_00000172e2bba210;
    %load/vec4 v00000172e2bb8a60_0;
    %flag_set/vec4 9;
    %flag_get/vec4 9;
    %jmp/0 T_2.2, 9;
    %load/vec4 v00000172e2bb8ba0_0;
    %inv;
    %and;
T_2.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_2.0, 8;
    %load/vec4 v00000172e2bb8f60_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 3;
    %cmp/u;
    %jmp/1 T_2.3, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 3;
    %cmp/u;
    %jmp/1 T_2.4, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 3;
    %cmp/u;
    %jmp/1 T_2.5, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 3;
    %cmp/u;
    %jmp/1 T_2.6, 6;
    %dup/vec4;
    %pushi/vec4 4, 0, 3;
    %cmp/u;
    %jmp/1 T_2.7, 6;
    %dup/vec4;
    %pushi/vec4 5, 0, 3;
    %cmp/u;
    %jmp/1 T_2.8, 6;
    %dup/vec4;
    %pushi/vec4 6, 0, 3;
    %cmp/u;
    %jmp/1 T_2.9, 6;
    %dup/vec4;
    %pushi/vec4 7, 0, 3;
    %cmp/u;
    %jmp/1 T_2.10, 6;
    %delay 1, 0;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8600_0, 0, 8;
    %jmp T_2.12;
T_2.3 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8600_0, 0, 8;
    %jmp T_2.12;
T_2.4 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8c40_0, 0, 8;
    %jmp T_2.12;
T_2.5 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8ce0_0, 0, 8;
    %jmp T_2.12;
T_2.6 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8e20_0, 0, 8;
    %jmp T_2.12;
T_2.7 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb9140_0, 0, 8;
    %jmp T_2.12;
T_2.8 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8ec0_0, 0, 8;
    %jmp T_2.12;
T_2.9 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8740_0, 0, 8;
    %jmp T_2.12;
T_2.10 ;
    %delay 1, 0;
    %load/vec4 v00000172e2bb90a0_0;
    %store/vec4 v00000172e2bb8380_0, 0, 8;
    %jmp T_2.12;
T_2.12 ;
    %pop/vec4 1;
T_2.0 ;
    %load/vec4 v00000172e2bb8ba0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_2.13, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8600_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8c40_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8ce0_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8e20_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb9140_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8ec0_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8740_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2bb8380_0, 0, 8;
T_2.13 ;
    %jmp T_2;
    .thread T_2;
    .scope S_00000172e2bce890;
T_3 ;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v00000172e2c23a60_0, 0, 8;
    %pushi/vec4 0, 0, 3;
    %store/vec4 v00000172e2c23ec0_0, 0, 3;
    %pushi/vec4 0, 0, 3;
    %store/vec4 v00000172e2c248c0_0, 0, 3;
    %pushi/vec4 1, 0, 3;
    %store/vec4 v00000172e2c25220_0, 0, 3;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000172e2c24640_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000172e2bb8420_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000172e2c245a0_0, 0, 1;
    %delay 3, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000172e2c24640_0, 0, 1;
    %pushi/vec4 10, 0, 8;
    %store/vec4 v00000172e2c23a60_0, 0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000172e2c245a0_0, 0, 1;
    %end;
    .thread T_3;
    .scope S_00000172e2bce890;
T_4 ;
    %delay 10, 0;
    %load/vec4 v00000172e2bb8420_0;
    %inv;
    %store/vec4 v00000172e2bb8420_0, 0, 1;
    %jmp T_4;
    .thread T_4;
    .scope S_00000172e2bce890;
T_5 ;
    %delay 50, 0;
    %vpi_call 2 27 "$finish" {0 0 0};
    %end;
    .thread T_5;
    .scope S_00000172e2bce890;
T_6 ;
    %vpi_call 2 30 "$monitor", $time, "%d | %d  %d | inA %d OutA %d %d | %d %d %d", v00000172e2c23a60_0, v00000172e2c24320_0, v00000172e2c24820_0, v00000172e2c23ec0_0, v00000172e2c248c0_0, v00000172e2c25220_0, v00000172e2c24640_0, v00000172e2bb8420_0, v00000172e2c245a0_0 {0 0 0};
    %end;
    .thread T_6;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "test1_tb.v";
    "./test1.v";
